hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestOmMetadataManager.java

```diff
@@ -666,4 +668,66 @@ public void testAllTablesAreProperInOMMetadataManagerImpl() {
 
     Assert.assertEquals(tablesByDefinition, tablesInManager);
   }
+
+  @Test
+  public void testListSnapshot() throws Exception {
+    String vol1 = "vol1";
+    String bucket1 = "bucket1";
+
+    OMRequestTestUtils.addVolumeToDB(vol1, omMetadataManager);
+    addBucketsToCache(vol1, bucket1);
+    String prefixSnapshotA = "key-a";
+    String prefixSnapshotB = "key-b";
+    TreeSet<String> snapshotsASet = new TreeSet<>();
+    TreeSet<String> snapshotsSet = new TreeSet<>();
+
+    for (int i = 1; i <= 100; i++) {
+      if (i % 2 == 0) {
+        snapshotsASet.add(prefixSnapshotA + i);
+        snapshotsSet.add(prefixSnapshotA + i);
+        OMRequestTestUtils.addSnapshotToTable(vol1, bucket1,
+            prefixSnapshotA + i, omMetadataManager);
+      } else {
+        snapshotsSet.add(prefixSnapshotB + i);
+        OMRequestTestUtils.addSnapshotToTableCache(vol1, bucket1,
+            prefixSnapshotB + i, omMetadataManager);
+      }
+    }
+
+    //Test listing snapshots with no volume name.
+    Assert.assertThrows(OMException.class, () -> omMetadataManager.listSnapshot(
+        null, null, null, null));
+
+    //Test listing snapshots with no bucket name.
+    Assert.assertThrows(OMException.class, () -> omMetadataManager.listSnapshot(
+        vol1, null, null, null));
+
+    //Test listing all snapshots which have prefix "key-a".
+    List<SnapshotInfo> snapshotInfos = omMetadataManager.listSnapshot(vol1,
+        bucket1, null, prefixSnapshotA);
+    Assert.assertEquals(snapshotInfos.size(), 50);
+    for (SnapshotInfo snapshotInfo : snapshotInfos) {
+      Assert.assertTrue(snapshotInfo.getName().startsWith(prefixSnapshotA));
+    }
+
+    //Test listing snapshots which have prefix "key-a" and start with "key-a10".
+    String startKey = prefixSnapshotA + "10";
+    snapshotInfos = omMetadataManager.listSnapshot(
+        vol1, bucket1, startKey, prefixSnapshotA);
+    Assert.assertEquals(snapshotsASet.tailSet(startKey).size() - 1,
+        snapshotInfos.size());
+    for (SnapshotInfo snapshotInfo : snapshotInfos) {
+      Assert.assertFalse(snapshotInfo.getName().equals(startKey));
```

Comment:
maybe this should be "greater than" instead of "not equals"?

Reply:
This assertion doen't need now.